applications/RANSApplication/custom_strategies/rans_fractional_step_strategy.h

```diff
@@ -0,0 +1,306 @@
+                    << "Fractional velocity converged in " << it + 1
+                    << " iterations." << std::endl;
+                break;
+            }
+        }
+
+        KRATOS_INFO_IF("RansFractionalStepStrategy", !Converged && BaseType::GetEchoLevel() > 0)
+            << "Fractional velocity iterations did not converge." << std::endl;
+
+        // Compute projections (for stabilization)
+        r_process_info.SetValue(FRACTIONAL_STEP, 4);
+        this->ComputeSplitOssProjections(rModelPart);
+
+        // 2. Pressure solution (store pressure variation in PRESSURE_OLD_IT)
+        r_process_info.SetValue(FRACTIONAL_STEP, 5);
+
+        const double eta = (r_process_info.Has(PRESSURE_COEFFICIENT))
+                               ? r_process_info[PRESSURE_COEFFICIENT]
+                               : 1.0;
+
+#pragma omp parallel for
+        for (int i_node = 0; i_node < n_nodes; ++i_node)
+        {
+            auto it_node = rModelPart.NodesBegin() + i_node;
+            const double old_press = it_node->FastGetSolutionStepValue(PRESSURE);
+            it_node->FastGetSolutionStepValue(PRESSURE_OLD_IT) = -eta * old_press;
+        }
+
+        KRATOS_INFO_IF("RansFractionalStepStrategy", BaseType::GetEchoLevel() > 0)
+            << "Calculating Pressure." << std::endl;
+        double NormDp = this->mpPressureStrategy->Solve();
+
+#pragma omp parallel for
+        for (int i_node = 0; i_node < n_nodes; ++i_node)
+        {
+            auto it_node = rModelPart.NodesBegin() + i_node;
+            it_node->FastGetSolutionStepValue(PRESSURE_OLD_IT) +=
+                it_node->FastGetSolutionStepValue(PRESSURE);
+        }
+
+        // 3. Compute end-of-step velocity
+        KRATOS_INFO_IF("RansFractionalStepStrategy", BaseType::GetEchoLevel() > 0)
+            << "Updating Velocity." << std::endl;
+        r_process_info.SetValue(FRACTIONAL_STEP, 6);
+
+        this->CalculateEndOfStepVelocity();
+
+        // Set the output tuple as the fractional velocity convergence and pressure norm
+        return std::make_tuple(Converged, NormDp);
+    }
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    /// Assignment operator.
+    RansFractionalStepStrategy& operator=(RansFractionalStepStrategy const& rOther)
+    {
+    }
+
+    /// Copy constructor.
+    RansFractionalStepStrategy(RansFractionalStepStrategy const& rOther)
+    {
+    }
```

Comment:
We can move these to the life cycle section and use the `delete` specifier to remove the private section of the file. Up to you.

Reply:
i like the delete :) thanks. it is changed now :+1: